data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/BulkRetryStrategy.java

```diff
@@ -261,9 +262,13 @@ private BulkOperationRequestResponse handleRetriesAndFailures(final Accumulating
     private void handleFailures(final AccumulatingBulkRequest<BulkOperationWrapper, BulkRequest> bulkRequest, final BulkResponse bulkResponse, final Throwable failure) {
         if (failure == null) {
             for (final BulkResponseItem bulkItemResponse : bulkResponse.items()) {
-                // Skip logging the error for version conflicts
-                if (bulkItemResponse.error() != null && !VERSION_CONFLICT_EXCEPTION_TYPE.equals(bulkItemResponse.error().type())) {
-                    LOG.warn("operation = {}, error = {}", bulkItemResponse.operationType(), bulkItemResponse.error().reason());
+                if(isItemInError(bulkItemResponse)) {
+                    // Skip logging the error for version conflicts
+                    final ErrorCause error = bulkItemResponse.error();
+                    if (error != null && VERSION_CONFLICT_EXCEPTION_TYPE.equals(error.type())) {
+                        break;
+                    }
+                    LOG.warn("operation = {}, status = {}, error = {}", bulkItemResponse.operationType(), bulkItemResponse.status(), error != null ? error.reason() : "");
                 }
             }
             handleFailures(bulkRequest, bulkResponse.items());
```

Comment:
Could we do a similar rest status check on line 290 as well for the BulkResponse?

Reply:
Good catch. Thanks!

Reply:
Are you referring to this line?

```
if (bulkResponse.errors()) {
```

I think you are suggesting that we also check the status of the HTTP response. I don't see any methods supporting this on this class. I believe that the client will throw an exception if the whole HTTP status is an error status. That's what this catch is for:

```
        } catch (Exception e) {
            incrementErrorCounters(e);
            return handleRetriesAndFailures(bulkRequestForRetry, retryCount, null, e);
        }
```

Reply:
Got it, makes sense